### PR TITLE
windows: fix some websocket tests

### LINKED ISSUE #8394: [windows] ServerWebSocket tests are timing out
```ts
bun-debug test test\js\bun\websocket\websocket-server.test.ts
```

```ts
bun test v1.0.25-canary.23 (f966445e)


Timeout: test "open" timed out after 1000ms
✗ Server > websocket > open [1999.34ms]
Timeout: test "close" timed out after 1000ms
✗ Server > websocket > close [1998.25ms]
Timeout: test "message" timed out after 1000ms
✗ Server > websocket > message [1998.86ms]
Timeout: test "drain" timed out after 1000ms
✗ Server > websocket > drain [1998.59ms]
Timeout: test "ping" timed out after 1000ms
✗ Server > websocket > ping [1998.90ms]
Timeout: test "pong" timed out after 1000ms
✗ Server > websocket > pong [1998.34ms]
Timeout: test "maxPayloadLength" timed out after 1000ms
✗ Server > websocket > maxPayloadLength [1998.33ms]
Timeout: test "backpressureLimit" timed out after 1000ms
✗ Server > websocket > backpressureLimit [1998.68ms]
✎ Server > websocket > closeOnBackpressureLimit
✎ Server > websocket > perMessageDeflate
Timeout: test "(default)" timed out after 1000ms
✗ ServerWebSocket > binaryType > (default) [1998.50ms]
Timeout: test "(invalid)" timed out after 1000ms
✗ ServerWebSocket > binaryType > (invalid) [1998.53ms]
Timeout: test "nodebuffer" timed out after 1000ms
✗ ServerWebSocket > binaryType > nodebuffer [1998.55ms]
Timeout: test "arraybuffer" timed out after 1000ms
✗ ServerWebSocket > binaryType > arraybuffer [1998.69ms]
Timeout: test "uint8array" timed out after 1000ms
✗ ServerWebSocket > binaryType > uint8array [1998.81ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > send() > string (ascii) [1998.53ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > send() > string (latin1) [1998.38ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > send() > string (utf-8) [1998.45ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > send() > Uint8Array (utf-8) [1998.52ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > send() > ArrayBuffer (utf-8) [1998.51ms]
Timeout: test "Buffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > send() > Buffer (utf-8) [1998.49ms]
Timeout: test "(benchmark)" timed out after 30000ms
✗ ServerWebSocket > send() > (benchmark) [59998.77ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > sendBinary() > Uint8Array (utf-8) [1996.64ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > sendBinary() > ArrayBuffer (utf-8) [1998.49ms]
Timeout: test "Buffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > sendBinary() > Buffer (utf-8) [1998.57ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > sendText() > string (ascii) [1998.46ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > sendText() > string (latin1) [1998.47ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > sendText() > string (utf-8) [1998.76ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > subscribe() > string (ascii) [1998.79ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > subscribe() > string (latin1) [1998.44ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > subscribe() > string (utf-8) [1998.47ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > publish() > string (ascii) [1998.54ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > publish() > string (latin1) [1998.62ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() > string (utf-8) [1998.49ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() > Uint8Array (utf-8) [1998.78ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() > ArrayBuffer (utf-8) [1998.58ms]
Timeout: test "Buffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() > Buffer (utf-8) [1998.48ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publishBinary() > Uint8Array (utf-8) [1998.82ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publishBinary() > ArrayBuffer (utf-8) [1998.48ms]
Timeout: test "Buffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publishBinary() > Buffer (utf-8) [1998.89ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > publishText() > string (ascii) [1998.46ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > publishText() > string (latin1) [1998.77ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publishText() > string (utf-8) [1998.57ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > publish() with { publishToSelf: true } > string (ascii) [1998.40ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > publish() with { publishToSelf: true } > string (latin1) [1998.68ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() with { publishToSelf: true } > string (utf-8) [1998.84ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() with { publishToSelf: true } > Uint8Array (utf-8) [1998.71ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() with { publishToSelf: true } > ArrayBuffer (utf-8) [1998.50ms]
Timeout: test "Buffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > publish() with { publishToSelf: true } > Buffer (utf-8) [1998.84ms]
Timeout: test "(no argument)" timed out after 1000ms
✗ ServerWebSocket > ping() > (no argument) [1998.68ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > ping() > string (ascii) [1998.85ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > ping() > string (latin1) [1998.80ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > ping() > string (utf-8) [1998.43ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > ping() > Uint8Array (utf-8) [1998.50ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > ping() > ArrayBuffer (utf-8) [1998.95ms]
Timeout: test "Buffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > ping() > Buffer (utf-8) [1998.42ms]
Timeout: test "(no argument)" timed out after 1000ms
✗ ServerWebSocket > pong() > (no argument) [1998.53ms]
Timeout: test "string (ascii)" timed out after 1000ms
✗ ServerWebSocket > pong() > string (ascii) [1998.66ms]
Timeout: test "string (latin1)" timed out after 1000ms
✗ ServerWebSocket > pong() > string (latin1) [1998.44ms]
Timeout: test "string (utf-8)" timed out after 1000ms
✗ ServerWebSocket > pong() > string (utf-8) [1998.80ms]
Timeout: test "Uint8Array (utf-8)" timed out after 1000ms
✗ ServerWebSocket > pong() > Uint8Array (utf-8) [1998.49ms]
Timeout: test "ArrayBuffer (utf-8)" timed out after 1000ms
✗ ServerWebSocket > pong() > ArrayBuffer (utf-8) [1998.54ms]
```

### PR DESCRIPTION
**Pull request by nektro**
- remove old loop check that doesn't exist anymore and was tripping an assert no longer true
- don't encode !isBinary data
- fix a assuming path separator is / bug

Fixes #8394

### CODE REVIEW

File: src/http/websocket_http_client.zig [hunk 1]
```diff
@@ -1636,12 +1622,12 @@ pub fn NewWebSocketClient(comptime ssl: bool) type {
             len: usize,
             op: u8,
         ) callconv(.C) void {
-            if (this.tcp.isClosed() or this.tcp.isShutdown()) {
+            if (this.tcp.isClosed() or this.tcp.isShutdown() or op > 0xF) {
```
Comment (Jarred-Sumner) [inline src/http/websocket_http_client.zig:1625]: @cirospaciari is this right

File: src/js/thirdparty/ws.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ class BunWebSocket extends EventEmitter {
       } else if (event === "message") {
         this.#ws.addEventListener("message", ({ data }) => {
           const isBinary = typeof data !== "string";
-          if (isBinary) {
```
Comment (Jarred-Sumner) [inline src/js/thirdparty/ws.js]: This feels suspicious

File: src/http/websocket_http_client.zig (hunk 1 above)
Comment (nektro) [reply src/http/websocket_http_client.zig:1625]: https://github.com/oven-sh/bun/blob/cff5dc0/src/http/websocket.zig
https://www.rfc-editor.org/rfc/rfc6455#section-5.2

**Comment by Jarred-Sumner** [comment]
Websocket tests are sigterm’ing

File: src/js/thirdparty/ws.js (hunk 2 above)
Comment (nektro) [reply src/js/thirdparty/ws.js]: `ws.test.ts` posix SIGTERMs are from this

**Comment by nektro** [comment]
I think this is good to go now

**Pull request merged**